### PR TITLE
[RDY] Sort out the modules

### PR DESCRIPTION
**Pull request by Hjdskes**
`shell.client` was mostly a console REPL, which should really be its own module considering we will also create an Eclipse client. It was thus extracted to `shell.client.console`, leaving only two interfaces into `shell.client` (`IEditor` and `IDisplay`). These were subsequently moved into `shell.core`, seeing each client uses these interfaces anyway. This removed the `shell.client` module entirely, after which `shell.client.console` was renamed to `shell.console`. I envision the Eclipse plugin to be named `shell.eclipse`.

### CODE REVIEW

File: shell.core/src/main/java/org/metaborg/spoofax/shell/core/CoreModule.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.metaborg.spoofax.shell.commands.SpoofaxEvaluationCommand;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Singleton;
+import com.google.inject.name.Names;
+
+/**
+ * Core library bindings.
+ */
+public class CoreModule extends AbstractModule {
```
Comment (Balletie) [inline shell.core/src/main/java/org/metaborg/spoofax/shell/core/CoreModule.java:15]: I don't see this Module used when I do Ctrl+f `CoreModule`.

Comment (Hjdskes) [inline shell.core/src/main/java/org/metaborg/spoofax/shell/core/CoreModule.java:15]: I have extended `CoreModule` with `ReplModule`.

Comment (Balletie) [inline shell.core/src/main/java/org/metaborg/spoofax/shell/core/CoreModule.java:15]: Does it work?

Comment (Balletie) [inline shell.core/src/main/java/org/metaborg/spoofax/shell/core/CoreModule.java:15]: It didn't, but now it's fixed by calling `super.configure()`

**Pull request merged**